test/native/libs/jninativestacks.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <jni.h>
+#include <math.h>
+
+JNIEXPORT double doCpuTask() {
+    int i = 0;
+    double result = 0;
+
+    while (i < 100000000) {
+        i++;
+        result += sqrt(i);
+        result += pow(i, sqrt(i));
+    }
+    return result;
+}
+
+__attribute__((unused)) JNIEXPORT double generateDeepStack(int count) {
+    char frame[0x20000];
+    // Prevent from being optimized by compiler
+    (void)frame;
+
+    if (count == 0) {
+        return doCpuTask();
+    } else {
+        return generateDeepStack(count - 1);
+    }
+}
+
+__attribute__((unused)) JNIEXPORT jdouble JNICALL Java_test_stackwalker_StackGenerator_generateLargeFrame(JNIEnv* env, jclass cls) {
```

Comment:
Naming could be better. Can be simply `largeFrame`, since these functions do not _generate_ frames.